src/sync/Synchroniser.js

```diff
@@ -81,6 +84,39 @@ export class Synchroniser {
     this.batchSize = MIN_SYNC_BATCH_SIZE;
   };
 
+  fetchWrapper = async (url, opts) => {
+    const { body } = opts;
+
+    const bugsnagNotify = (message, response) => {
+      bugsnagClient.notify(new Error(message), {
+        context: 'SYNC ERROR',
+        metaData: { url, body, responseText: response.text() },
+      });
+    };
+    // eslint-disable-next-line no-undef
+    const response = await fetch(url, opts);
+    if (response.status < 200 || response.status >= 300) {
```

Comment:
I'm not really a fan of having the generic fetch method handling all errors, especially when it is hard to change in the future, like this. I think it makes pinpointing exactly where errors are occurring a little harder as there are just errors bubbling up all over the place. I think the more generic error catching - json parsing etc and bugsnag notifying are good, but aspects which can differ request to request I think are better by passing parameters i.e. valid status', or having the caller take care of it. Although that is probably over doing things for such a simple use case

Reply:
I do agree that some http request require non generic error handling, but all of the request in Synchroniser.js have the same pattern.

As you've mentioned, it is an overkill for caller to specify exactly what to validate.

By the way generic fetch wrappers are very common